content/articles/2019-01-tibble-2-0-0.Rmarkdown

```diff
@@ -0,0 +1,187 @@
+iris %>%
+  view()
+```
+
+
+The `enframe()` function always has been a good way to convert a (named) vector to a two-column data frame.
+In this version, conversion to a one-column data frame is also supported by setting the `name` argument to `NULL`.
+This is now the recommended way to turn a vector to a one-column tibble, due to changes to the default implementation of `as_tibble()`.
+
+```{r enframe}
+enframe(letters[1:3])
+enframe(letters[1:3], name = NULL)
+```
+
+## Changes that affect package developers
+
+All functions have been assigned a lifecycle.
+The _tibble_ package has now reached the "stable" lifecycle, functions in a different lifecycle stage are marked as such in their documentation.
+One example is the `add_row()` function: it is unclear if it  should ensure that all columns have length one by wrapping in a list if necessary, adn a better implementation is perhaps possible once _tibble_ uses the _vctrs_ package, see below.
+Therefore this function is marked "questioning".
+Learn more about lifecycle in the tidyverse at https://www.tidyverse.org/lifecycle/.
+
+The new `.rows` argument to `tibble()` and `as_tibble()` allows specifying the expected number of rows explicitly, even if it's evident from the data.
+This supports writing even more defensive code.
+The `nrow` argument to the low-level `new_tibble()` constructor is now mandatory, on the other hand most expensive checks have been moved to the new `validate_tibble()` function.
+This means that constructions of tibbles is now faster by default if you know that the inputs are correct, but you can always double-check if needed.
+
+```{r nrow}
+tibble(a = 1, b = 1:3, .rows = 3)
+tibble(a = 1, b = 2:3, .rows = 3)
+tibble(a = 1, .rows = 3)
+as_tibble(iris[1:3, ], .rows = 3)
+new_tibble(list(a = 1:3), nrow = 3)
+bad <- new_tibble(list(a = 1:2), nrow = 3)
+validate_tibble(bad)
+```
+
+Column name repair has more direct support, via the new `.name_repair` argument to `tibble()` and `as_tibble()`.
+It takes the following values:
+
+  - `"minimal"`: No name repair or checks, beyond basic existence.
+  - `"unique"`: Make sure names are unique and not empty.
+  - `"check_unique"`: (default value), no name repair, but check they are `unique`.
+  - `"universal"`: Make the names `unique` and syntactic.
+  - a function: apply custom name repair (e.g., `.name_repair = make.names` or `.name_repair = ~make.names(., unique = TRUE)` for names in the style of base R).
+
+```{r repair-names}
+## by default, duplicate names are not allowed
+tibble(` ` = 1, ` ` = 2)
+## you can authorize duplicate names
+tibble(` ` = 1, ` ` = 2, .name_repair = "minimal")
+## or request that the names be made unique
+tibble(` ` = 1, ` ` = 2, .name_repair = "unique")
+## or or universal
+tibble(` ` = 1, ` ` = 2, .name_repair = "universal")
+```
+
+Row name handling is stricter.
+Row names are never (and never were) supported in `tibble()` and `new_tibble()`, and are now stripped by default in `as_tibble()`.
```

Comment:
```suggestion
Row names are not (and never were) supported in `tibble()` and `new_tibble()`, and are now stripped by default in `as_tibble()`.
```

Reply:
I wanted to say that no way of using `tibble()` or `new_tibble()` would yield a tibble with row names.?

Reply:
Something is wrong with "are never" - maybe should be "were never"?